src/gpu/sycl/binary_kernels.hpp

```diff
@@ -36,14 +36,24 @@ struct binary_kernel_vec_t {
             xpu::sycl::in_memory_arg_t &src0, xpu::sycl::in_memory_arg_t &src1,
             xpu::sycl::out_memory_arg_t &dst,
             xpu::sycl::in_memory_arg_t &src0_scale,
-            xpu::sycl::in_memory_arg_t &src1_scale, data_type_t scales_dt)
+            xpu::sycl::in_memory_arg_t &src1_scale, data_type_t scales_dt,
+            xpu::sycl::in_memory_arg_t &po1_src,
+            xpu::sycl::in_memory_arg_t &po2_src,
+            xpu::sycl::in_memory_arg_t &po3_src,
+            xpu::sycl::in_memory_arg_t &po4_src,
+            xpu::sycl::in_memory_arg_t &po5_src)
```

Comment:
Could you change that to a variadic number of arguments through a `vector` or variadic functions?

Reply:
I can not pass a std::vector to a kernel. I could do a std::tuple, but it would only change this interface without removing the limitation of maximum 5 post ops. So I would prefer not to, to make it more obvious there is this limitation.

Reply:
Could you, please, share where the limitation is coming from?

Reply:
Since the kernels are not templated on the post ops we need to limit the maximal number of them (this is 5 for sycl kernels, even before this PR) and transfer data for each potential post op to the kernel (whether or not it is actually used in the kernel). You can see this in the design of `sycl_post_ops_t` struct.

Reply:
IIUC, we could use std::vector or std::array as parameter. However explicit copy of the vector/array elements would have to happen before kernel call right ?